net-analyzer/suricata/suricata-7.0.6.ebuild

```diff
@@ -0,0 +1,223 @@
+"
+
+LICENSE="GPL-2"
+SLOT="0/7"
+KEYWORDS="~amd64 ~riscv ~x86"
+IUSE="+af-packet af-xdp bpf control-socket cuda debug +detection geoip hardened hyperscan lua lz4 nflog +nfqueue redis systemd test"
+VERIFY_SIG_OPENPGP_KEY_PATH="/usr/share/openpgp-keys/openinfosecfoundation.org.asc"
+
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}
+	af-xdp? ( bpf )
+	bpf? ( af-packet )
+	lua? ( ${LUA_REQUIRED_USE} )
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	acct-group/suricata
+	acct-user/suricata
+	dev-libs/jansson:=
+	dev-libs/libpcre2
+	dev-libs/libyaml
+	net-libs/libnet:*
+	net-libs/libnfnetlink
+	dev-libs/nspr
+	dev-libs/nss
+	$(python_gen_cond_dep 'dev-python/pyyaml[${PYTHON_USEDEP}]')
+	>=net-libs/libhtp-0.5.48
+	net-libs/libpcap
+	sys-apps/file
+	sys-libs/libcap-ng
+	af-xdp?	( net-libs/xdp-tools )
+	bpf?       ( dev-libs/libbpf )
+	cuda?      ( dev-util/nvidia-cuda-toolkit )
+	geoip?     ( dev-libs/libmaxminddb:= )
+	hyperscan? ( dev-libs/vectorscan:= )
+	lua?       ( ${LUA_DEPS} )
+	lz4?       ( app-arch/lz4 )
+	nflog?     ( net-libs/libnetfilter_log )
+	nfqueue?   ( net-libs/libnetfilter_queue )
+	redis?     ( dev-libs/hiredis:= )
+"
+DEPEND="${RDEPEND}
+	>=dev-build/autoconf-2.69-r5
+	virtual/rust
+"
+BDEPEND="verify-sig? ( >=sec-keys/openpgp-keys-oisf-20200807 )"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.0.1_configure-no-lz4-automagic.patch"
+	"${FILESDIR}/${PN}-5.0.7_configure-no-hyperscan-automagic.patch"
+	"${FILESDIR}/${PN}-6.0.0_default-config.patch"
+	"${FILESDIR}/${PN}-7.0.2_configure-no-sphinx-pdflatex-automagic.patch"
+	"${FILESDIR}/${PN}-7.0.5_configure-fortify_source.patch"
+)
+
+pkg_pretend() {
+	if use af-xdp && use kernel_linux && kernel_is -lt 4 18; then
+		ewarn "Kernel 4.18 or newer is required for AF_XDP"
+	fi
+
+	if use bpf && use kernel_linux; then
+		if kernel_is -lt 4 15; then
+			ewarn "Kernel 4.15 or newer is necessary to use all XDP features like the CPU redirect map"
+		fi
+
+		CONFIG_CHECK="~XDP_SOCKETS"
+		ERROR_XDP_SOCKETS="CONFIG_XDP_SOCKETS is not set, making it impossible for Suricata to load XDP programs. "
+		ERROR_XDP_SOCKETS+="Other eBPF features should work normally."
+		check_extra_config
+	fi
+}
+
+src_prepare() {
+	default
+	sed -ie 's/docdir =.*/docdir = ${datarootdir}\/doc\/'${PF}'\//' "${S}/doc/Makefile.am" || die
+	eautoreconf
+}
+
+src_configure() {
+	# Bug #861242
+	filter-lto
+
+	local myeconfargs=(
+		--localstatedir=/var
+		--runstatedir=/run
+		--enable-non-bundled-htp
+		--enable-gccmarch-native=no
+		--enable-python
+		--disable-coccinelle
+		$(use_enable af-packet)
+		$(use_enable af-xdp)
+		$(use_enable bpf ebpf)
+		$(use_enable control-socket unix-socket)
+		$(use_enable cuda)
+		$(use_enable detection)
+		$(use_enable geoip)
+		$(use_enable hardened gccprotect)
+		$(use_enable hardened pie)
+		$(use_enable hyperscan)
+		$(use_enable lz4)
+		$(use_enable nflog)
+		$(use_enable nfqueue)
+		$(use_enable redis hiredis)
+		$(use_enable test unittests)
```

Comment:
There are new configure options by upstream in this version. `ja3` and `ja4`. Could you check whether they're useful to us in any way, make them use flags, or disable them unconditionally with `--disable-ja` options?